admin/includes/functions/general.php

```diff
@@ -3683,12 +3683,16 @@ function zen_get_ip_address() {
   /**
    * sanitize for validity as an IPv4 or IPv6 address
    */
-  $ip = preg_replace('~[^a-fA-F0-9.:%/]~', '', $ip);
+    $original_ip = $ip;
+    $ip = filter_var((string)$ip, FILTER_VALIDATE_IP, FILTER_FLAG_IPV6 | FILTER_FLAG_IPV4);
```

Comment:
Why limit this to only `FILTER_FLAG_IPV6` and `FILTER_FLAG_IPV4` such that when an additional IPV designation is generated this code would need to be modified?

Reply:
Because then it's a known set of flags (remembering the fuss when the defaults for `htmlspecialchars` changed.

Reply:
Is there another IPV spec nearing release?

Reply:
> Is there another IPV spec nearing release?

Not in a position to know and do not know of one. Looking at the ways to maximize the use and accessibility of this new method without having to touch the code again. Ie. use of an `$options` variable that can be overridden to serve the desire/need of the code section.

For example to prevent a private or reserved range ip address from being used as a capability not as a code change, or to use a reverse logic to only allow a private or reserved range ip address to be returned/used.

> Because then it's a known set of flags (remembering the fuss when the defaults for `htmlspecialchars` changed. 

So then why not assign the value to be returned if the value does not pass the filter?  Doesn't leaving this to the current default also lead to the same concern/confusion in that if the default changes from `false` to perhaps `0`, then it again won't work.

Reply:
Because the point of filtering the IP address to a known type is to ensure that the space reserved in the database to hold the value is sufficient.  You are overthinking the requirements here, IMO.